src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -7177,169 +7206,274 @@ internal static List<CompletionResult> CompleteStatementFlags(TokenKind kind, st
+            // If we fail to resolve the type of any key we return null
+            for (int i = nestedKeys.Count - 1; i >= 0; i--)
+            {
+                if (currentType is null)
+                {
+                    return null;
+                }
+
+                var typeMembers = typeContext.GetMembersByInferredType(currentType, false, null);
+                currentType = null;
+                foreach (var member in typeMembers)
+                {
+                    if (member is PropertyInfo propertyInfo)
+                    {
+                        if (propertyInfo.Name.Equals(nestedKeys[i], StringComparison.OrdinalIgnoreCase))
+                        {
+                            currentType = new PSTypeName(propertyInfo.PropertyType);
+                            break;
+                        }
+                    }
+                    else if (member is PropertyMemberAst memberAst && memberAst.Name.Equals(nestedKeys[i], StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (memberAst.PropertyType is null)
+                        {
+                            return null;
+                        }
+                        else
+                        {
+                            currentType = new PSTypeName(memberAst.PropertyType.TypeName);
+                        }
+                        break;
+                    }
+                }
+            }
+
+            return currentType;
+        }
+
         internal static List<CompletionResult> CompleteHashtableKey(CompletionContext completionContext, HashtableAst hashtableAst)
         {
+            Ast previousAst = hashtableAst;
+            Ast parentAst = hashtableAst.Parent;
+            string parameterName = null;
+            var nestedHashtableKeys = new List<string>();
+
+            // This loop determines if it's a nested hashtable and what the outermost hashtable is used for (Dynamic keyword, command argument, etc.)
+            while (parentAst is not null)
+            {
+                if (parentAst is HashtableAst parentTable)
+                {
+                    foreach (var pair in parentTable.KeyValuePairs)
+                    {
+                        if (pair.Item2 == previousAst)
+                        {
+                            // Try to get the value of the hashtable key in the nested hashtable.
+                            // If we fail to get the value then return early because we can't generate any useful completions
+                            object value;
+                            if (SafeExprEvaluator.TrySafeEval(pair.Item1, completionContext.ExecutionContext, out value))
+                            {
+                                var stringValue = value as string;
+                                if (stringValue is null)
+                                {
+                                    return null;
+                                }
```

Comment:
Minor suggestion
```suggestion
                                if (value is string stringValue)
                                {
                                    return null;
                                }
```

Reply:
Oh, typo in my suggestion. I meant
```
if (value is not string stringValue)
{
    return null
}
```
But it's up to you.

Reply:
Why no? Sure?

Reply:
@iSazonov What? I have applied the updated suggestion if that's what you are asking.